### PR TITLE
Add upgrade guide for v12

### PR DESCRIPTION
**Pull request by kmaehashi**

### CODE REVIEW

File: docs/source/upgrade.rst [hunk 1]
```diff
@@ -5,6 +5,58 @@ Upgrade Guide
...
 Please see also the :ref:`compatibility_matrix` for supported environments of each major version.
 
+CuPy v12
+========
+
+Change in :class:`cupy.cuda.Device` Behavior
+--------------------------------------------
+
+The CUDA current device (set via :meth:`cupy.cuda.Device.use()` or ``cudaSetDevice()``) will be reactivated when exiting a device context manager.
+This is a revert of the :ref:`change introduced in CuPy v10 <change in CuPy Device behavior>`, making the behavior identical to the one in CuPy v9 or earlier.
```
Comment (emcastillo) [inline docs/source/upgrade.rst:15]:
```suggestion
This reverts the :ref:`change introduced in CuPy v10 <change in CuPy Device behavior>`, making the behavior identical to the one in CuPy v9 or earlier.
```

**Review by emcastillo — APPROVED** [review]
LGTM! just a nitpick

**Comment by kmaehashi** [comment]
Test passed, preview looks fine to me:
https://cupy--7430.org.readthedocs.build/en/7430/upgrade.html#cupy-v12

**Pull request merged**